src/transformers/data/processors/utils.py

```diff
@@ -16,16 +16,20 @@
 
 import copy
 import csv
+import dataclasses
 import json
 import logging
+from dataclasses import dataclass
+from typing import Optional
 
 from ...file_utils import is_tf_available, is_torch_available
 
 
 logger = logging.getLogger(__name__)
 
 
-class InputExample(object):
+@dataclass(frozen=True)
```

Comment:
Immutability (frozen=True) is a nice side effect of using this.

Reply:
bug in run_tf_glue.py
```
Traceback (most recent call last):
  File "run_tf_glue.py", line 51, in <module>
    train_dataset = glue_convert_examples_to_features(data["train"], tokenizer, 128, TASK)
  File "/Users/ljzhao/huggingface/lib/python3.7/site-packages/transformers/data/processors/glue.py", line 86, in glue_convert_examples_to_features
    example = processor.tfds_map(example)
  File "/Users/ljzhao/huggingface/lib/python3.7/site-packages/transformers/data/processors/utils.py", line 115, in tfds_map
    example.label = self.get_labels()[int(example.label)]
  File "<string>", line 3, in __setattr__
dataclasses.FrozenInstanceError: cannot assign to field 'label'
```
solved by setting frozen=False

Reply:
Hmm, good catch @egrcc, thanks for reporting. @jplu do you need mutability of `InputExample`s in `run_tf_glue`?

Reply:
or can we do it another way?

Reply:
Unfortunately in this case (I mean the way the label is computed) you cannot really do something else than unfrozing the examples. A better solution might be to keep the dataclass frozen but computing the label in same time than we create the example itself and not with the features.

Reply:
Ok got it. I'll unfreeze it for now, but yeah I think we should probably store the "featurized" version in there.